modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3250CrossTenantSubscriptionTestCase.java

```diff
@@ -0,0 +1,225 @@
+    public static final String DEFAULT_APPLICATION = "DefaultApplication";
+    private WebDriver driver;
+
+    private static final Log log = LogFactory.getLog(APIMANAGER3250CrossTenantSubscriptionTestCase.class);
+
+    private String TEST_DATA_TENANT_FIRST_NAME = "admin",
+            TEST_DATA_TENANT_LAST_NAME = "admin",
+            TEST_DATA_ADMIN_USER_NAME = "admin",
+            TEST_DATA_PASSWORD = "123456",
+            TEST_DATA_API_NAME = "testAPI",
+            TEST_DATA_API_END_POINT = "http://localhost:8080/api",
+            TEST_DATA_API_VERSION = "1.0.0";
+
+    @BeforeClass(alwaysRun = true)
+    protected void init() throws Exception {
+        super.init();
+        driver = BrowserManager.getWebDriver();
+    }
+
+
+    public void generateTenant(String postfix) throws Exception {
+
+        WebDriverWait wait = new WebDriverWait(driver, 10);
+        // wait until load the page
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("#menu-panel-button3 > span")));
+
+        driver.findElement(By.cssSelector("#menu-panel-button3 > span")).click();
+        driver.findElement(By.linkText("Add New Tenant")).click();
+        // wait until load the page
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("domain")));
+
+        driver.findElement(By.id("domain")).clear();
+        driver.findElement(By.id("domain")).sendKeys("test" + postfix + ".com");
+        driver.findElement(By.id("admin-firstname")).clear();
+        driver.findElement(By.id("admin-firstname")).sendKeys(TEST_DATA_TENANT_FIRST_NAME);
+        driver.findElement(By.id("admin-lastname")).clear();
+        driver.findElement(By.id("admin-lastname")).sendKeys(TEST_DATA_TENANT_LAST_NAME);
+        driver.findElement(By.id("admin")).clear();
+        driver.findElement(By.id("admin")).sendKeys(TEST_DATA_ADMIN_USER_NAME);
+        driver.findElement(By.id("admin-password")).clear();
+        driver.findElement(By.id("admin-password")).sendKeys(TEST_DATA_PASSWORD);
+        driver.findElement(By.id("admin-password-repeat")).clear();
+        driver.findElement(By.id("admin-password-repeat")).sendKeys(TEST_DATA_PASSWORD);
+        driver.findElement(By.id("admin-email")).clear();
+        driver.findElement(By.id("admin-email")).sendKeys("admin@test" + postfix + ".com");
+        driver.findElement(By.cssSelector("input.button")).click();
+        driver.findElement(By.cssSelector("button[type=\"button\"]")).click();
+    }
+
+    @Test(groups = "wso2.am")
+    public void checkCrossTenantSubscription() throws Exception {
+
+        WebDriverWait wait = new WebDriverWait(driver, 10);
+
+
+        driver.get(getLoginURL());
+
+        // wait until load the page
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("txtUserName")));
+
+        driver.findElement(By.id("txtUserName")).clear();
+        driver.findElement(By.id("txtUserName")).sendKeys("admin");
```

Comment:
don't hard code admin user name and password. You can retrieve these details from automation context